File: app/src/main/java/com/bigscreen/mangindo/network/ConnectionManager.java
package com.bigscreen.mangindo.network;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;


public class ConnectionManager {

    public static int TYPE_WIFI = 1;

    public static int TYPE_MOBILE = 2;

    public static int TYPE_NOT_CONNECTED = 0;

    private Context context;

    public ConnectionManager(Context context) {
        this.context = context;
    }

    public int getConnectivityStatus() {
        ConnectivityManager connectivityManager = (ConnectivityManager) context
                .getSystemService(Context.CONNECTIVITY_SERVICE);

        NetworkInfo activeNetwork = connectivityManager.getActiveNetworkInfo();
        if (activeNetwork != null && activeNetwork.isConnected() && activeNetwork.isAvailable()) {
            if (activeNetwork.getType() == ConnectivityManager.TYPE_WIFI) {
                return TYPE_WIFI;
            }

            if (activeNetwork.getType() == ConnectivityManager.TYPE_MOBILE) {
                return TYPE_MOBILE;
            }
        }
        return TYPE_NOT_CONNECTED;
    }

    public boolean isConnected() {
        return getConnectivityStatus() != TYPE_NOT_CONNECTED;
    }
}

File: app/src/main/java/com/bigscreen/mangindo/network/ConnectionManager.kt
package com.bigscreen.mangindo.network

import android.content.Context
import android.net.ConnectivityManager

class ConnectionManager(private val context: Context) {

    companion object {
        const val TYPE_WIFI = 1
        const val TYPE_MOBILE = 2
        const val TYPE_NOT_CONNECTED = 0
    }

    fun getConnectivityStatus(): Int {
        val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val activeNetwork = connectivityManager.activeNetworkInfo
        if (activeNetwork != null && activeNetwork.isConnected && activeNetwork.isAvailable) {
            if (activeNetwork.type == ConnectivityManager.TYPE_WIFI) {
                return TYPE_WIFI
            }
            if (activeNetwork.type == ConnectivityManager.TYPE_MOBILE) {
                return TYPE_MOBILE
            }
        }
        return TYPE_NOT_CONNECTED
    }

    fun isConnected(): Boolean {
        return getConnectivityStatus() != TYPE_NOT_CONNECTED
    }
}
